### ISSUE TITLE
<style> tags in response are location dependent

### ISSUE THREAD

**Issue by geoffrey-eisenbarth** [bigskysoftware/htmx#778]
It appears that if a `<style>` tag is the first tag of a response, then it is not properly swapped into place. I've only gotten my `<style>` tags to work when including an OOB Swap before the `<style>`. 

I believe the following should reproduce the issue:

## HTML ##
```
<html>
  <body>
    <style id="style">
      .example {
        color: red;
      }
    </style>
  
    <h2 class="example">Example</h2>
    <h3 id="oob-swap">Initial Text</h3>
    <button
      hx-get="/endpoint_works/"
      hx-target="#style"
      hx-swap="outerHTML"
    >
      Click me! (Works)
    </button>
    <button
      hx-get="/endpoint_broken/"
      hx-target="#style"
      hx-swap="outerHTML"
    >
      Click me! (Broken)
    </button>
  </body>
</html>
```

## Server (Django) ##
**views.py**
```
from django.http import HttpRequest, HttpResponse


def endpoint_works(request: HttpRequest) -> HttpResponse:
  html = """
    <h3 id="oob-swap" hx-swap-oob="outerHTML">New Text</h3>
    <style id='style'>
       .example { 
         color: blue;
       }
    </style>
  """
  return HttpResponse(html)
  
def endpoint_broken(request: HttpRequest) -> HttpResponse:
  html = """
    <style id='style'>
       .example { 
         color: blue;
       }
    </style>
    <h3 id="oob-swap" hx-swap-oob="outerHTML">New Text</h3>
  """
  return HttpResponse(html)
```
  
**urls.py**
```
from django.urls import path
from .views import endpoint_works, endpoint_broken


urlpatterns = [
  path(
    '/endpoint_works/',
    endpoint_works,
    name='works',
  ),
  path(
    '/endpoint_broken/',
    endpoint_broken,
    name='broken',
  ),
]
```

**Comment by geoffrey-eisenbarth** [comment]
@1cg tagging this per your twitter post today.

Sometimes I just want to swap a `<style>`, but I have to add in a throw-away element before it so that it will render. 

Not sure if this would be considered important to others, but 🤷

**Comment by geoffrey-eisenbarth** [comment]
After messing around a bit more, I think the issue is related to doing an outerHTML swap for <style> tags. If I just do innerHTMK swap, it works as expected.

**Comment by 1cg** [comment]
I would bet that if you used template parsing (rather than the standard IE-compatible parsing) it would work.  Going to close this as a corner case that would be too complex to fix pre htmx 2.0 (when we will drop IE support)

**Comment by geoffrey-eisenbarth** [comment]
All good. Like I mentioned, my work-around of using `innerHTML` swap works good enough for me.
